src/factories/meta/CollectionCollector_factory.h

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Simon Gardner
+
+#include "extensions/jana/JOmniFactory.h"
+#include "algorithms/meta/CollectionCollector.h"
+
+namespace eicrecon {
+
+    template<class T>
+    class CollectionCollector_factory : public JOmniFactory<CollectionCollector_factory<T>> {
+public:
+    using AlgoT = eicrecon::CollectionCollector<typename T::collection_type>;
+
+private:
+    std::unique_ptr<AlgoT> m_algo;
+
+    typename JOmniFactory<CollectionCollector_factory<T>>::template VariadicPodioInput<T> m_inputs {this};
+    typename JOmniFactory<CollectionCollector_factory<T>>::template PodioOutput<T> m_output {this};
+
+public:
+
+    void Configure() {
+        m_algo = std::make_unique<AlgoT>(this->GetPrefix());
+        m_algo->init(this->logger());
+    }
+
+    void ChangeRun(int64_t run_number) {
+    }
+
+    void Process(int64_t run_number, uint64_t event_number) {
+
+        try {
+            std::vector<gsl::not_null<const typename T::collection_type*>>  in_collections;
+            for (const auto& in_collection : m_inputs()) {
+                in_collections.push_back(gsl::not_null<const typename T::collection_type*>{in_collection});
+            }
+            typename T::collection_type* merged_collection = m_output().get();
+            m_algo->process(in_collections, merged_collection);
+        }
+        catch(std::exception &e) {
+            throw JException(e.what());
```

Comment:
For debugging:
```suggestion
            this->logger()->error(e.what());
            throw JException(e.what());
```
I would not be surprised if this was a case of an exception getting silently eaten by JANA2, likely an exception in the `for` loop over `m_inputs()` that then results in not filling `merged_collection` at all.

Reply:
On my machine and events, for some currently undiagnosed reason running with this combination works:
SiBarrelTrackerRecHits
SiBarrelVertexRecHits
SiEndcapTrackerRecHits
TOFBarrelRecHit          
MPGDBarrelRecHits
MPGDDIRCRecHits       
B0TrackerRecHits

Adding TOFEndcapRecHits, starts calling the factory Process function then segfaults after a few events

Adding any of the below, the factory Process function is never called.
OuterMPGDBarrelRecHits
BackwardMPGDEndcapRecHits
ForwardMPGDEndcapRecHits

Reply:
While I hate suggesting it, for now we should probably just try-catch and print `e.what()` as a warning, check if that works in the scope of this PR, and dump the underlying issue on the plate of the TOF and MPGD group as an issue.

Reply:
I think this is a case for the optional collections, if any of them are missing the factory doesn't run. `e.what()` was never getting called when missing collections were requested.

It appears I was running on an old simulation file where different MPGD collections existed, looking at a new one the MPGD collections that broke my local run are present whereas MPGDDIRCHits doesn't appear to be. Can you confirm this?

Hopefully the TOF problems are related to the older simulation file too.

Reply:
MPGD DIRC is for epic_arches, we don't need to support it anymore. I would remove `try catch`, if possible.

Reply:
If the reconstruction needs to work seamlessly with multiple detector configurations we'll need to allow for these to be optional variadic inputs. That does however allow the input list to get cluttered with unsupported collections. @nathanwbrei is this possible yet?

The list at the moment contains all of the detectors from a/b/c but will only work if they're all there. A separate issue but tangentially related, I hadn't realised the 18x275.xml was still a brycecanyon configuration and all of the CI is done using craterlake.

Reply:
"No hope" is overstating it quite a bit. Go ahead and merge what you have now, when I add optional collection support I'll update this at the same time.